gadget/offset.go

```diff
@@ -0,0 +1,83 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package gadget
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+)
+
+// OffsetWriter implements support for writing the start offsets of structure
+// and its content at locations defined by offset-write property. structures and
+// their content.
+type OffsetWriter struct {
+	ps         *PositionedStructure
+	sectorSize Size
+}
+
+func asLBA(value, sectorSize Size) uint32 {
+	return uint32(value / sectorSize)
+}
+
+func deployOffsetWrite(out io.WriteSeeker, offset Size, value uint32) error {
```

Comment:
Nitpick: in the kernel things like that are called `pwrite`, perhaps we could mimic the name?